sample-apps/prometheus/metrics_collector.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type metricCollector struct {
+	counters   []prometheus.Counter
+	gauges     []prometheus.Gauge
+	histograms []prometheus.Histogram
+	summarys   []prometheus.Summary
+
+	metricCount int
+}
+
+func newMetricCollector() metricCollector {
+	return metricCollector{}
+}
+
+func (mc *metricCollector) updateMetrics() {
+	for {
+		// generate new metrics in 30 second intervals
+		time.Sleep(time.Second * 30)
```

Comment:
Want to at the end of the loop instead? So the user always sees some metrics as soon as they launch the program.